### PR TITLE
feat: initial commit to create cli for teamlink

### PR DESCRIPTION
**Pull request by sailorlqh**

### CODE REVIEW

File: pkg/cli/root.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"context"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+// rootCmd defines the starting command structure.
+var rootCmd = func() cli.Command {
+	return &cli.RootCommand{
+		Name: "tlclt",
```
Comment (Celdir) [inline pkg/cli/root.go:27]: nit typo: "tlclt" -> "tlctl"

Comment (sailorlqh) [reply pkg/cli/root.go]: Thank you! Fixed!

File: pkg/cli/sync.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-system-auth-token",
+		Target:  &c.sourceToken,
+		Aliases: []string{"st", "src-token"},
+		Usage:   `Token to authenticate with source system to read membership information`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "des-system-auth-token",
```
Comment (Celdir) [inline pkg/cli/sync.go:111]: I would say "dest" or "dst" for naming consistency

File: pkg/cli/sync.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	if _, ok := slices.BinarySearch(allowedSourceSystem, strings.ToUpper(c.source)); !ok {
```
Comment (Celdir) [inline pkg/cli/sync.go:131]: `BinarySearch` requires `allowedSourceSystem` to be in sorted order, and when we add more source systems in the future I'm not sure that other devs will remember to keep that list sorted. 
Also, I doubt the number of source or destination systems will ever be greater than ~10, so I think using linear search (`slices.Contains`) would be the better choice for simplicity.

File: pkg/cli/sync.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+		Target:  &c.destination,
+		Aliases: []string{"d", "dst"},
+		Example: "GitHub",
+		Usage:   `The target system for membership information`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "group-mapping-config",
+		Target:  &c.groupMappingConfig,
+		Aliases: []string{"g", "gc", "group-config", "g-config"},
```
Comment (drevell) [inline pkg/cli/sync.go:88]: Why so many aliases?

File: pkg/cli/sync.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		Example: "group-mapping-config.textproto",
+		Usage: `The group mapping config that contains group id mapping ` +
+			`from source system to destination system`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "user-mapping-config",
+		Target:  &c.userMappingConfig,
+		Aliases: []string{"u", "uc", "user-config", "u-config"},
```
Comment (drevell) [inline pkg/cli/sync.go:97]: Here too

File: pkg/cli/sync.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+var (
+	_                        cli.Command = (*SyncCommand)(nil)
+	allowedSourceSystem                  = []string{"GOOGLEGROUP"}
```
Comment (drevell) [inline pkg/cli/sync.go:28]: Should this be "groups" instead of "group"? It's not a singular group right? It's the name of the service, "Google Groups"

Comment (drevell) [inline pkg/cli/sync.go:28]: The capitalization seems a little unusual, is there a reason?

File: pkg/cli/sync.go (hunk 3 above)
Comment (drevell) [reply pkg/cli/sync.go:131]: +1 for slices.Contains

**Review by drevell — APPROVED** [review]

File: pkg/cli/sync.go (hunk 6 above)
Comment (sailorlqh) [reply pkg/cli/sync.go]: I am using `ToUpper` to check if input is in the allowlist, so I used upper case here. Do you think using lower case will be better?

Comment (sailorlqh) [reply pkg/cli/sync.go]: Updated to `GOOGLEGROUPS`

File: pkg/cli/sync.go (hunk 4 above)
Comment (sailorlqh) [reply pkg/cli/sync.go]: I was thinking to provide users more choices, now I think it's unnecessary. Removed, and only kept `gc` and `uc`

File: pkg/cli/sync.go (hunk 3 above)
Comment (sailorlqh) [reply pkg/cli/sync.go]: Fixed. Thank you!

File: pkg/cli/sync.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Sync membership information from source and target system.
+
+  Sync membership from google group to GitHub
+
+  tlctl sync run \
+	-src GoogleGroup \
+	-dest GitHub \
```
Comment (Celdir) [inline pkg/cli/sync.go:57]: Help string args should be updated to reflect current names/aliases, e.g. "dest" -> "dst"

Comment (Celdir) [reply pkg/cli/sync.go:57]: Also are auth tokens required for source and destination systems? If so the example command should include those flags

**Review by Celdir — APPROVED** [review]

File: pkg/cli/sync.go (hunk 7 above)
Comment (sailorlqh) [reply pkg/cli/sync.go]: Updated. Thank you for catching this!

**Pull request merged**